File: 2022/day1/solution.py
with open("input.txt") as f:
    elves_bags = f.read().splitlines()

calories_tracker: list[int] = []
current_calories = 0
for calories in elves_bags:
    if calories == '':
        calories_tracker.append(current_calories)
        current_calories = 0
        continue

    current_calories += int(calories)

# Part 1
print(max(calories_tracker))

#Part 2
print(sum(sorted(calories_tracker, reverse=True)[:3]))

File: 2022/day1/solution.ts
const input = await Deno.readTextFile("input.txt");
const elves_bags = input.split("\n");

const calories_tracker = [];
let current_calories = 0;

for (let i = 0; i < elves_bags.length; i++) {
    const calories = elves_bags[i];
    if (calories == "") {
        calories_tracker.push(current_calories);
        current_calories = 0;
        continue;
    }

    current_calories += parseInt(calories);
}

calories_tracker.sort((a, b) => b - a);

// Part 1
console.log(calories_tracker[0]);

// Part 2
console.log(calories_tracker.slice(0, 3).reduce((a, b) => a + b, 0));
